components/feed/DeleteButton.tsx

```diff
@@ -0,0 +1,47 @@
+"use client";
+
+import { useRouter } from "next/navigation";
+
+import deleteFeedItem from "@/lib/feed/deleteFeedItem";
+
+import type { FeedItemType, FeedCommentType } from ".";
+
+const DeletePostButton = ({
+  postData,
+  commentData,
+}: {
+  postData: FeedItemType;
+  commentData?: FeedCommentType;
+}) => {
+  const router = useRouter();
+  const handleDelete = async (e: React.MouseEvent) => {
+    e.preventDefault();
+    if (typeof commentData !== "undefined") {
```

Comment:
```suggestion
    if (!!commentData) {
```

Reply:
Actually you don't even need this if statement because the `deleteFeedItem` func checks it